src/routes/Admin/AdminView.js

```diff
@@ -13,31 +18,171 @@ class Admin extends Component {
     this.state = {
       products: [],
       isLading: false,
+      showModal: false,
+      showingLoadForms: false,
     };
-    this.clickItem = this.clickItem.bind(this);
+
+    this.navigateToItem = this.navigateToItem.bind(this);
+    this.handleEdit = this.handleEdit.bind(this);
+    this.handleOpenModal = this.handleOpenModal.bind(this);
+    this.handleCloseModal = this.handleCloseModal.bind(this);
+    this.handleCloseModalQuick = this.handleCloseModalQuick.bind(this);
+    this.deleteItem = this.deleteItem.bind(this);
+    this.createAddItemGlobal = this.createAddItemGlobal.bind(this);
+    this.modalCreate = this.modalCreate.bind(this);
   }
 
   async componentDidMount() {
     this.setState({ isLading: true });
     const productsJson = await fetch('/api/v1/products');
+    // console.log('need inf', productsJson)
     const products = await productsJson.json();
     this.setState({ products, isLading: false });
   }
 
-  clickItem = (evt, id) => {
+  handleOpenModal() {
+    this.setState({
+      showModal: true,
+      propsItem: undefined,
+      createNewItem: true,
+    });
+  }
+
+  handleCloseModal() {
+    this.setState({
+      showingLoadForms: true,
+      // id: '',
+      // title: '',
+      // description: '',
+      // price: '',
+    });
+    setTimeout(() => {
+      this.setState({
+        showingLoadForms: false,
+        showModal: false,
+      });
+    }, 2000);
+  }
+
+  handleCloseModalQuick() {
+    this.setState({
+      showModal: false,
+      // id: '',
+      // title: '',
+      // description: '',
+      // price: '',
+    });
+  }
+
+  navigateToItem = (evt, id) => {
     this.props.router.push(`/admin/product/${id}`);
     // browserHistory.push(`/admin/product/${id}`);
   }
 
+  handleEdit = (propsItem) => {
+    this.setState({
+      showModal: true,
+      propsItem,
+    });
+  }
+
+  deleteItem(id) {
+    fetch(`/api/v1/products/${id}`, {
+      method: 'DELETE',
+    })
+      .then(res => res.json())
+      .catch((error) => {
+        console.log('Request failed', error);
+      })
+      .then(this.setState({ products: this.state.products.filter(i => i.id !== id) }));
```

Comment:
the same problem, it should be in the then callback above, definitely not after the catch.